### DISCUSSION TITLE
RS485 and termination

### DISCUSSION THREAD

**Discussion by brownjd** [aqualinkd/AqualinkD#168] [Q&A] [answered]
I currently have an RS4 with both red network blocks used. One set of wires comes from an aqualink receiver. I am not sure about where the wires goes. 

Is it okay to simply double up and insert the A/B wires from the FTDI adapter into the Yellow/Black terminals along with the existing wiring? Or does it need its own dedicated terminals? 

How does termination figure in here? Is that what the Red/Green 1/4 terminals are for? Should you connect all 4 wires (wiki only references the data wires). How should you terminate when the FTDI documentation indicates it has internal termination?

**Comment by sfeakes** [comment]
You can piggyback wiring.  But when doing that, you need to be careful with termination, there is no standard so you’ll need to look at what devices are terminated.  You’ll usually be better off extending the wiring from client devise (Ie keypad). Since no standard and even manufacturers change termination between versions it something you’ll need to play with.  You only need the 2 data wires for a USB receiver to work since it has its own power.

**Comment by brownjd** [comment]
I can’t tell that anything is terminated. There are just two sets of four wires going in to the red blocks.

I don’t have any keypads. It is all app based. I think the second set of wires might go to the heater or something.

I tried the two wires (orange and yellow) based on the FTDI doc. I tried reversing them, but I can’t get your test commend to echo back anything from the jandy controller.

How do you go about terminating? And do you need to alter the existing terminations if they exist?

Jason

**Comment by sfeakes** [comment]
Where does each set of wires go?  When you say app based, what do you use to control the pool / change settings, if iAqualink, then there is also a transmitter on the RS485 that acts like a keypad.  You might be better of connecting to the posts on the transmitter.  Either way, use the AqualinkD serial logger and connect the wires at either the control panel or iAqualink transmitter, the serial will usually tell you if you have errors on the bus when receiving messages.

**Comment by brownjd** [comment]
One set of wires does go to the  iAquaLink transmitter. I think the second set might go to either the heater or maybe the variable speed motor.

I did connect the wires to the control side and tried the echo command in the wiki, but got nothing. I’ll try enabling the serial logger and see if I get anything.

Jason

**Comment by brownjd** [comment]
I disconnected the iAqualink so that I would have a free block. I then connected two wires, orange and yellow (per the FTDI data sheet) to 2 and 3. I tried twisting the termination wires together and leaving them untwisted. Didn't seem to make a difference. 

I also tried pairing them with the orange and yellow so that I had green and yellow in #3 and brown and orange in #2. That looked promising as it does seem to hang the serial logger. It never prints anything out though. However the "sudo -s eval '\''stty raw -echo < /dev/ttyUSB0; cat -vte /dev/ttyUSB0'\'" will print out the following:

M-|M-LM-`M-}^L<M-~M-^@M-xA^]M-~?M-^?M-pM-|^C

Attached is a picture of the setup, the dmesg output and the serial logger. Any suggestions?

dmesg  |  grep usb
[    0.275859] usbcore: registered new interface driver usbfs
[    0.276061] usbcore: registered new interface driver hub
[    0.276227] usbcore: registered new device driver usb
[    0.276785] usb_phy_generic phy: supply vcc not found, using dummy regulator
[    2.595905] usbcore: registered new interface driver smsc95xx
[    3.608982] dwc_otg 20980000.usb: DWC OTG Controller
[    3.612277] dwc_otg 20980000.usb: new USB bus registered, assigned bus number 1
[    3.615616] dwc_otg 20980000.usb: irq 56, io mem 0x00000000
[    3.625646] usb usb1: New USB device found, idVendor=1d6b, idProduct=0002, bcdDevice= 5.15
[    3.632025] usb usb1: New USB device strings: Mfr=3, Product=2, SerialNumber=1
[    3.635383] usb usb1: Product: DWC OTG Controller
[    3.638669] usb usb1: Manufacturer: Linux 5.15.32+ dwc_otg_hcd
[    3.642054] usb usb1: SerialNumber: 20980000.usb
[    3.655841] usbcore: registered new interface driver usb-storage
[    3.688454] usbcore: registered new interface driver usbhid
[    3.691979] usbhid: USB HID core driver
[    4.066565] usb 1-1: new high-speed USB device number 2 using dwc_otg
[    4.361840] usb 1-1: New USB device found, idVendor=0424, idProduct=9514, bcdDevice= 2.00
[    4.361910] usb 1-1: New USB device strings: Mfr=0, Product=0, SerialNumber=0
[    4.681879] usb 1-1.1: new high-speed USB device number 3 using dwc_otg
[    4.813090] usb 1-1.1: New USB device found, idVendor=0424, idProduct=ec00, bcdDevice= 2.00
[    4.813141] usb 1-1.1: New USB device strings: Mfr=0, Product=0, SerialNumber=0
[    4.998951] SMSC LAN8700 usb-001:003:01: attached PHY driver (mii_bus:phy_addr=usb-001:003:01, irq=POLL)
[    5.020521] smsc95xx 1-1.1:1.0 eth0: register 'smsc95xx' at usb-20980000.usb-1.1, smsc95xx USB 2.0 Ethernet, b8:27:eb:a3:2d:69
[    5.141067] usb 1-1.2: new high-speed USB device number 4 using dwc_otg
[    5.318334] usb 1-1.2: New USB device found, idVendor=148f, idProduct=5370, bcdDevice= 1.01
[    5.334831] usb 1-1.2: New USB device strings: Mfr=1, Product=2, SerialNumber=3
[    5.346379] usb 1-1.2: Product: 802.11 n WLAN
[    5.354917] usb 1-1.2: Manufacturer: Ralink
[    5.363095] usb 1-1.2: SerialNumber: 1.0
[    5.473349] usb 1-1.4: new full-speed USB device number 5 using dwc_otg
[    5.636125] usb 1-1.4: New USB device found, idVendor=0403, idProduct=6001, bcdDevice= 6.00
[    5.652289] usb 1-1.4: New USB device strings: Mfr=1, Product=2, SerialNumber=3
[    5.663707] usb 1-1.4: Product: TTL232R
[    5.671556] usb 1-1.4: Manufacturer: FTDI
[    5.679440] usb 1-1.4: SerialNumber: FT5YDUHZ
[   25.366335] usbcore: registered new interface driver usbserial_generic
[   25.366546] usbserial: USB Serial support registered for generic
[   26.111873] usbcore: registered new interface driver ftdi_sio
[   26.112066] usbserial: USB Serial support registered for FTDI USB Serial Device
[   26.113141] usb 1-1.4: Detected FT232RL
[   26.268340] usb 1-1.4: FTDI USB Serial Device converter now attached to ttyUSB0
[   38.033601] usb 1-1.2: reset high-speed USB device number 4 using dwc_otg
[   39.647238] usbcore: registered new interface driver rt2800usb
[ 1080.452302] ftdi_sio ttyUSB0: usb_serial_generic_read_bulk_callback - urb stopped: -32
[ 1080.461325] ftdi_sio ttyUSB0: usb_serial_generic_read_bulk_callback - urb stopped: -32
[ 1080.557625] usb 1-1.4: USB disconnect, device number 5
[ 1094.930237] usb 1-1.5: new full-speed USB device number 6 using dwc_otg
[ 1095.085033] usb 1-1.5: New USB device found, idVendor=0403, idProduct=6001, bcdDevice= 6.00
[ 1095.085103] usb 1-1.5: New USB device strings: Mfr=1, Product=2, SerialNumber=3
[ 1095.085138] usb 1-1.5: Product: TTL232R
[ 1095.085162] usb 1-1.5: Manufacturer: FTDI
[ 1095.085185] usb 1-1.5: SerialNumber: FT5YDUHZ
[ 1095.106067] usb 1-1.5: Detected FT232RL
[ 1095.109374] usb 1-1.5: FTDI USB Serial Device converter now attached to ttyUSB0


AqualinkD serial_logger V1.4
Notice:  RS Serial: Logging serial information!
Error:   RS Serial: ERROR, Timeout on serial port, nothing read! Please check /dev/ttyUSB0
Please wait./

Notice:  RS Serial: RS485 interface received 0 packets in 25 seconds (~0.00 Msg/Sec)
Notice:  RS Serial: Jandy Control Panel Model   : 
Notice:  RS Serial: Jandy Control Panel Version : 
Notice:  RS Serial: Jandy ID's found
Notice:  RS Serial: 

![2337594D-EE61-47D3-B7FF-261E84ACE9EA_1_102_o](https://user-images.githubusercontent.com/867810/163890631-edcc013b-7da0-4378-9b9f-2764b30106b2.jpeg)

**Comment by sfeakes** [comment]
Can you post a link to the RS485 to USB device you are using.  It looks like it’s set to RS232 and not RS485, and also looks like TTL but in the wrong direction.  It also looks like it’s initiating it multiple times.  Your dmesg is very odd.
I’d expect an output similar to below.
‘’’

0.958989] usb usb1: New USB device found, idVendor=1d6b, idProduct=0002
[    0.961718] usb usb1: New USB device strings: Mfr=3, Product=2, SerialNumber=1
[    0.964351] usb usb1: Product: DWC OTG Controller
[    0.966911] usb usb1: Manufacturer: Linux 4.9.67+ dwc_otg_hcd
[    0.969470] usb usb1: SerialNumber: 20980000.usb
[    0.973214] hub 1-0:1.0: USB hub found
[    0.979680] usbcore: registered new interface driver usb-storage
[    1.016979] usbcore: registered new interface driver usbhid
[    1.019780] usbhid: USB HID core driver
[    1.400919] usb 1-1: new full-speed USB device number 2 using dwc_otg
[    1.637387] usb 1-1: New USB device found, idVendor=0403, idProduct=6001
[    1.640564] usb 1-1: New USB device strings: Mfr=1, Product=2, SerialNumber=3
[    1.643492] usb 1-1: Product: FT232R USB UART
[    1.646310] usb 1-1: Manufacturer: FTDI
[    1.649139] usb 1-1: SerialNumber: A50285BI
[    9.819165] usbcore: registered new interface driver brcmfmac
[   11.184546] usbcore: registered new interface driver usbserial
[   11.184693] usbcore: registered new interface driver usbserial_generic
[   11.184814] usbserial: USB Serial support registered for generic
[   11.419705] usbcore: registered new interface driver ftdi_sio
[   11.419851] usbserial: USB Serial support registered for FTDI USB Serial Device
[   11.420157] ftdi_sio 1-1:1.0: FTDI USB Serial Device converter detected
[   11.420457] usb 1-1: Detected FT232RL
[   11.533103] usb 1-1: FTDI USB Serial Device converter now attached to ttyUSB0

‘’’

**Comment by brownjd** [comment]
FTDI USB-RS232-WE-1800-BT-5.0 Cable, USB to RS232 Serial, 1.8M, Wire END https://www.amazon.com/dp/B01HI9CXG4/ref=cm_sw_r_cp_api_i_XDCFE62027AD27FT2MH7?_encoding=UTF8&psc=1

Jason

**Comment by sfeakes** [comment, marked as answer]
This is the one you need in that pigtail style.
https://www.amazon.com/gp/product/B073RCSYX1/ref=ppx_yo_dt_b_search_asin_title?ie=UTF8&psc=1

The one you picked is UART RS232 more for programming an Arduino style chip / ESPxxx

**Comment by brownjd** [comment]
<sigh>

Thanks for your patience.
